docs/developers_guide.md

```diff
@@ -219,3 +219,109 @@ We plan to keep our code loosely up to date with the upstream Firecracker reposi
 to speed up testing environment setup requiring Kubernetes.
 
 * Current [eStargz](https://github.com/containerd/stargz-snapshotter) version is 0.13.0.
+
+
+## GPU Support
+
+Currently, vHive has not integrated with GPU, but the GPU support can be tested with stock Knative setups. Here are the steps to test Nvidia GPU support with stock Knative.
+
+### Install CUDA Drivers
+
+```
+./scripts/gpu/setup_cudadriver.sh
+sudo reboot -h now
+```
+
+### Install Stock Kubernetes and Knative
+
+Follow the guide to [setup stock knative](#testing-stock-knative-setup-or-images). 
+
+``` bash
+./scripts/cloudlab/setup_node.sh stock-only
+```
+
+### Install Nvidia Container Toolkit and Modify Containerd Settings
+
+You can use the script provided if the install of containerd is using  our script or manually edit the containerd settings following [Nvidia's official document](https://docs.nvidia.com/datacenter/cloud-native/kubernetes/install-k8s.html#install-nvidia-container-toolkit-nvidia-docker2).
+
+``` bash
+./scripts/gpu/setup_nvidiacontainerd.sh
+```
+
+
+### Start Containerd and Knative
+
+``` bash
+sudo screen -dmS containerd containerd; sleep 5;
+./scripts/cluster/create_one_node_cluster.sh stock-only
+```
+
+### Install NVIDIA Device Plugin
+
+Install the Nvidia Device plugin after all pods are running or completed.
+
+``` bash
+./scripts/gpu/setup_nvidiadevice.sh
+```
+At this point, all pods should be successfully deployed.
+``` bash
+$ kubectl get pods -A
+NAMESPACE              NAME                                       READY   STATUS      RESTARTS   AGE
+nvidia-device-plugin   nvidia-device-plugin-1684892866-rtgk9      1/1     Running     0          80s
+
+```
+And you can use the gpu-pod.yaml to test whether the Nvidia Device Plugin works or not
+
+``` bash
+kubectl apply -f ./scripts/gpu/gpu-pod.yaml
+```
+After it completed, check the log and it should be
+``` bash
+$ kubectl logs gpu-operator-test
+```
+``` bash
+[Vector addition of 50000 elements]
+Copy input data from the host memory to the CUDA device
+CUDA kernel launch with 196 blocks of 256 threads
+Copy output data from the CUDA device to the host memory
+Test PASSED
+Done
+```
+### Test Knative with GPU
+Using GPU in Knative is simple and same to the method in Kubernetes. The only change is to add the limits of GPU under the containers part.
+``` yaml
+resources:
+  limits:
+    nvidia.com/gpu: 1
+```
+You can also try deploying our container, which is a simple Go service that calls and returns the output of nvidia-smi.
+``` bash
+kn service apply -f ./scripts/gpu/gpu-service.yaml
```

Comment:
is it a service or a Knative function? we need only the function example

Reply:
It's a service. For Knative function version, it needs extra dependency for kn func. I'll try to change it to a function example